Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,304 @@
+
+                var trueNodeCondition = exceptionObject;
+                var falseNodeCondition = new UnopExpression(
+                    UnopExpression.UnopKind.LNot, trueNodeCondition, null);
+
+                // Create exception handling branch. True if exception is caught. False otherwise.
+                (var exceptionTrueNode, var exceptionFalseNode) =
+                    CreateExceptionBranchConditionNodes(state,
+                                                        trueNodeCondition,
+                                                        falseNodeCondition);
+
+                exceptionTrueNode.Instructions.Insert(0, exceptionAllocationCall);
+                exceptionFalseNode.Instructions.Insert(0, exceptionAllocationCall);
+
+                var catchVarIdentifier = state.GetIdentifier(Identifier.IdentKind.Normal);
+                var catchVarStore = new Store(catchVariable,
+                                              new VarExpression(returnVariable),
+                                              expressionType,
+                                              state.CurrentLocation);
+                exceptionTrueNode.Instructions.Add(catchVarStore);
+
+                RegisterNode(state, exceptionTrueNode);
+                RegisterNode(state, exceptionFalseNode);
+
+                Instruction nextInstructionIfNoException = null;
+
+                foreach (var singleInstruction in state.Method.Body.Instructions)
+                {
+                    if (singleInstruction.Offset >= endBlockOffset + 1)
+                    {
+                        nextInstructionIfNoException = singleInstruction;
+                        break;
+                    }
+                }
+                state.PushExpr(new ExnExpression(new VarExpression(returnVariable)), expressionType);
+
+                if (nextInstructionIfNoException != null)
+                {
+                    state.PushInstruction(nextInstructionIfNoException, exceptionFalseNode);
+                }
+
+                state.PushInstruction(instruction, exceptionTrueNode);
+                (instruction, _) = state.PopInstruction();
+            }
+            // Construct a finally block when unwrapped exception type is "System.Object".
+            else
+            {
+                state.PushRetExpr();
+                ParseCilInstruction(instruction, state);
+                state.PopInstruction();
+                state.PushInstruction(instruction.Next);
+
+                state.PushInstruction(instruction, exceptionHandlerNode);
+                (instruction, _) = state.PopInstruction();
+            }
+
+            /* Load caught exception variable. For example:
+            
+            node 4: Preds:2 Succs:6 EXN: 
+            n$25=*&CatchVar65:java.lang.Object*;
+            *&e:java.lang.Object*=n$25;*/
+            var newNode = new StatementNode(location: state.CurrentLocation,
```

Comment:
For the finally case, shouldn't the type of the node be FinallyBranch? 

![image](https://user-images.githubusercontent.com/38902361/123128703-3806f280-d400-11eb-88c8-7e49af93a308.png)

Reply:
Yeah, I saw this node kind exists, but never saw it in an example. So I have no idea when this node kind will happen. If you can find an example in Java, maybe I can add it in our translation.